app/controllers/vm_common.rb

```diff
@@ -1051,6 +1055,26 @@ def scan_histories
     end
   end
 
+  def parent_folder_id(vm)
+    if vm.orphaned
+      "xx-orph"
+    elsif vm.archived
+      "xx-arch"
+    elsif vm.cloud && vm.template
+      TreeBuilder.build_node_cid(vm.ext_management_system)
```

Comment:
This seems to be called only 1 time for discovery. I may be mistaken

But good point. we already know the base class and the parent id. This is possibly looking up an object from an id to then just paste together the object's base class (known) with the id (known)